### ISSUE TITLE
Tasks integration: Failing tests?

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/vscode-python-tox#26]
Tests seem to fail after merging the tasks integration: https://github.com/The-Compiler/vscode-python-tox/runs/7218607915?check_suite_focus=true

```
  1) ToxTaskProvider Test Suite
       getting tox tasks:
AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
0 !== 3
+ expected- actual
-0
+3
```

not sure what's going on there? I saw this error once locally, but when rerunning it went away, so I didn't think much of it...

@dariuszparys do you have any idea what could be going on there?

(follow-up to #5/#12)

**Comment by The-Compiler** [comment]
After 2 hours of debugging I think I finally got to the bottom of this and figured out how to make those tests run :sweat_smile: 

So the problem is that it's not possible to enter a new workspace in tests, because VS Code restarts the extension host when that's done, and thus outright refuses to do so when running tests:

- https://github.com/microsoft/vscode/issues/78740
- https://github.com/microsoft/vscode/issues/69335

When running the tests for the first time, that's confirmed by an error:

> Extension 'python-tox' failed to update workspace folders: Error: Entering a new workspace is not possible in tests.

Triggered from here:

https://github.com/The-Compiler/vscode-python-tox/blob/3bc01593ee1fc709b4bd5ee96abf6a16cb7ca8c4/src/test/suite/toxTaskProvider.test.ts#L22

And indeed the tests fail because the task runner can't find any workspace - `vscode.workspace.workspaceFolders` is empty here:

https://github.com/The-Compiler/vscode-python-tox/blob/46ecf1f27b9870dc04e685c1ae223edffb6e9028/src/toxTaskProvider.ts#L78-L80

However, the workspace state seems to get stored in `.vscode-test/` despite the error! Thus, when launching the tests a *second* time, they just happen to open `allenvs` as a workspace and everything works, until `.vscode-test/` is deleted again... Of course, the CI always starts with a clean state, which is why they probably always passed for @dariuszparys locally, but always fail on the CI. To top it all off, apparently I never ever enabled CI for PRs and didn't even notice until now... :hand_over_mouth: 

In 46ecf1f27b9870dc04e685c1ae223edffb6e9028, I now ended up changing `runTest.js` to open all examples as folders in a multi-folder workspace - with that setup, everything works again. Not perfect since it means tests aren't isolated as nicely from each other anymore, but better than nothing I guess...
